kf_lib_data_ingest/etl/stage_analyses.py

```diff
@@ -78,61 +81,63 @@ def compare_counts(
     messages = [title + '\n' + '='*len(title)]
     passed = True
 
-    one_keys = discovery_sources_one.keys()
-    two_keys = discovery_sources_two.keys()
+    col_list_one = list(discovery_sources_one.keys())
+    col_list_two = list(discovery_sources_two.keys())
+    comparison_df, diff_count = _compare(name_one, col_list_one,
+                                         name_two, col_list_two)
 
-    comparison_df, diff_num = _compare(name_one, one_keys, name_two, two_keys)
-
-    if diff_num != 0:
+    if diff_count != 0:
         msg = f'❌ Column names not equal between {name_one} and {name_two}'
         messages.extend(_format(msg, comparison_df))
         passed = False
 
-    for k in (one_keys | two_keys):
-        one_k_keys = discovery_sources_one.get(k, {}).keys()
-        two_k_keys = discovery_sources_two.get(k, {}).keys()
-
-        comparison_df, diff_num = _compare(name_one, one_k_keys,
-                                           name_two, two_k_keys)
+    for k in set(col_list_one + col_list_two):
+        comparison_df, diff_count = _compare(
+            name_one, [key for key in discovery_sources_one.get(k, {}).keys()
+                       if pandas.notnull(key)],
+            name_two, [key for key in discovery_sources_two.get(k, {}).keys()
+                       if pandas.notnull(key)]
+        )
 
-        if diff_num != 0:
+        if diff_count != 0:
             msg = (
                 f'❌ Column values for {k} not equal between {name_one} and '
                 f'{name_two}\n'
-                f'Number of different values = {diff_num}'
+                f'Number of different unique values = {diff_count}'
             )
             messages.extend(_format(msg, comparison_df))
             passed = False
 
     return passed, messages
 
 
-def _compare(name_one, list_one, name_two, list_two):
-    indicator = 'Errors'
-    one = pandas.DataFrame({name_one: list(list_one)}).dropna()
-    two = pandas.DataFrame({name_two: list(list_two)}).dropna()
-
+def _compare(
+    name_one, list_one, name_two, list_two
+):
+    indicator = 'Match'
+    one = pandas.DataFrame({name_one: list_one})
+    two = pandas.DataFrame({name_two: list_two})
     comparison_df = pandas.merge(one, two,
                                  left_on=name_one,
                                  right_on=name_two,
                                  how='outer', indicator=indicator)
-    diff_num = comparison_df[comparison_df[indicator] != 'both'].shape[0]
+    diff_count = comparison_df[comparison_df[indicator] != 'both'].shape[0]
 
     comparison_df[indicator].replace({
         'left_only': '❌',
         'right_only': '❌',
-        'both': ''
+        'both': '✅'
```

Comment:
I really like seeing both the good vs bad with emojis

Reply:
I figured fewer indicators was better, but ok